pkg/endpoint/policy.go

```diff
@@ -303,19 +322,23 @@ func (e *Endpoint) regeneratePolicy(owner Owner) (bool, error) {
 		policyChanged = true
 	}
 
-	log.Debugf("[%s] Done regenerating policyChanged=%v optsChanged=%v",
-		e.PolicyID(), policyChanged, optsChanged)
+	log.Debugf("[%s] Done regenerating revision=%v policyChanged=%v optsChanged=%v",
+		e.PolicyID(), revision, policyChanged, optsChanged)
 
-	// If no policy change occurred for this endpoint then the endpoint is
+	// If no policy or options change occurred for this endpoint then the endpoint is
 	// already running the latest revision, otherwise we have to wait for
 	// the regeneration of the endpoint to complete.
-	if !policyChanged {
+	if !(policyChanged || optsChanged) {
 		e.policyRevision = revision
+		e.nextPolicyRevision = revision
```

Comment:
You can update `nextPolicyRevision` outside of the branch

Reply:
Right!